src/extension.ts

```diff
@@ -131,35 +135,102 @@ const toSeverity = (severity: ValeSeverity): DiagnosticSeverity => {
 const toDiagnostic = (error: IValeErrorJSON): Diagnostic => {
     // vale prints one-based locations but code wants zero-based, so adjust
     // accordingly
-    const range = new Range(error.Line - 1, error.Span[0] - 1,
-        error.Line - 1, error.Span[1]);
-    const message = error.Link ?
-        `${error.Message} (${error.Check}, see ${error.Link})` :
-        `${error.Message} (${error.Check})`;
+    const range = new Range(
+        error.Line - 1,
+        error.Span[0] - 1,
+        error.Line - 1,
+        error.Span[1],
+    );
+    const message = error.Link
+        ? `${error.Message} (${error.Check}, see ${error.Link})`
+        : `${error.Message} (${error.Check})`;
     const diagnostic = new Diagnostic(
-        range, message, toSeverity(error.Severity));
+        range,
+        message,
+        toSeverity(error.Severity),
+    );
     diagnostic.source = "vale";
     diagnostic.code = error.Check;
     return diagnostic;
 };
 
-const readBinaryLocation = () => {
+/**
+ * Get the version of vale.
+ *
+ * @return A promise with the vale version string as single element.  If vale
+ * doesn't exist or if the version wasn't found the promise is rejected.
+ */
+const getValeVersion = async (
+    workspaceFolder?: WorkspaceFolder,
+): Promise<string> => {
+    const binaryLocation = readBinaryLocation(workspaceFolder);
```

Comment:
Reading the binary location now requires access to the `WorkspaceFolder`, which is derived by the `TextDocument`:

```ts
const workspaceFolder = workspace.getWorkspaceFolder(document.uri);
```

The `TextDocument` is only available (to my understanding) in the workspace events (e.g. `onDidOpenTextDocument`).